src/Lean/Util/Trace.lean

```diff
@@ -150,18 +150,27 @@ private def addTraceNode (oldTraces : PersistentArray TraceElem)
   let msg ← addMessageContext msg
   addTraceNodeCore oldTraces cls ref msg collapsed
 
-def withTraceNode [MonadExcept ε m] (cls : Name) (msg : Except ε α → m MessageData) (k : m α)
+def withSeconds [Monad m] [MonadLiftT BaseIO m] (act : m α) : m (α × Float) := do
+  let start ← IO.monoNanosNow
+  let a ← act
+  let stop ← IO.monoNanosNow
+  return (a, (stop - start).toFloat / 1000000000)
+
+def withTraceNode [MonadExcept ε m] [MonadLiftT BaseIO m] (cls : Name) (msg : Except ε α → m MessageData) (k : m α)
     (collapsed := true) : m α := do
   if !(← isTracingEnabledFor cls) then
     k
   else
     let ref ← getRef
     let oldTraces ← getResetTraces
-    let res ← observing k
-    addTraceNode oldTraces cls ref (← msg res) collapsed
+    let (res, secs) ← withSeconds <| observing k
```

Comment:
`withSeconds` is being executed even when `profiler` is set to false. Have you checked the impact on performance?

Reply:
I see you tried `!bench!` last week, but it seems it did not work.